### PR TITLE
adding cluster name to alert message and dedup_key

### PR DESCRIPTION
**Pull request by alavrovinfb**
added ability to add cluster name to alert message and dedup_key via CR environment key

### CODE REVIEW

File: deploy/examples/dashboards/SimpleDashboard.yaml [hunk 1]
```diff
@@ -6,6 +6,7 @@ metadata:
     app: grafana
 spec:
   name: simple-dashboard.json
+  environment: env-4
```
Comment (Evgeniy-L) [inline deploy/examples/dashboards/SimpleDashboard.yaml:9]: How will we deploy it on other env? Does it mean that we will need to modify dashboard for every env?

Comment (alavrovinfb) [reply deploy/examples/dashboards/SimpleDashboard.yaml:9]: This is the sample dashboard CR k8s manifest, we currently deploy dashboards via helm charts. In helm charts, we can use variables that can be substituted by a particular value from `values.yaml`, something like `environment: {{ .Values.env }}`.
But, we have to modify the dashboard helm chart.

File: deploy/crds/Grafana.yaml [hunk 2]
```diff
@@ -136,5 +136,16 @@ spec:
...
-              description: Reference to DB password secret
+              description: SecretKeySelector selects a key of a Secret.
+              properties:
+                key:
+                  description: The key of the secret to select from.  Must be a valid
+                    secret key.
+                  type: string
+                name:
+                  description: 'Name of the referent. More info: https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#names
+                    TODO: Add other useful fields. apiVersion, kind, uid?'
```
Comment (Evgeniy-L) [inline deploy/crds/Grafana.yaml:147]: can we remove this TODO?

**Review by Evgeniy-L — APPROVED** [review]
Looks ok

**Review by Evgeniy-L — APPROVED** [review]

**Comment by drewwells** [comment]
Please link to issue, what issue was this fixing? We should be layering our changes on top of the opensource project. These changes make our operator fork API incompatible with upstream grafana-operator.

**Comment by Evgeniy-L** [comment]
@drewwells, there was a general recommendation from management not to point to any internal issues in open source repos

**Pull request merged**